### PR TITLE
Make use of ExtendedMetadataBuilder in the Metrics extension

### LINKED ISSUE #7875: Make use of the new ExtendedMetadataBuilder in the Metrics extension
SR Metrics 2.4.1 introduced a builder for `ExtendedMetadata` to get rid of the constructor hell needed to construct `ExtendedMetadata` objects

### PR DESCRIPTION
**Pull request by jmartisk**
Fixes #7875

### CODE REVIEW

File: extensions/smallrye-metrics/runtime/src/main/java/io/quarkus/smallrye/metrics/runtime/SmallRyeMetricsRecorder.java [hunk 1]
```diff
@@ -558,28 +559,38 @@ private void micrometerJvmGcMetrics(MetricRegistry registry, ShutdownContext shu
...
-                    "Incremented for an increase in the size of the young generation memory pool after one GC to before the next",
-                    true,
-                    "jvm_gc_memory_allocated_bytes_total"), new LambdaCounter(gcMetrics::getAllocatedBytes));
+            registry.register(new ExtendedMetadataBuilder()
+                    .withName("jvm.gc.max.data.size")
+                    .withType(MetricType.GAUGE)
+                    .withUnit(MetricUnits.BYTES)
+                    .withDescription("Max size of old generation memory pool")
+                    .skipsScopeInOpenMetricsExportCompletely(true)
+                    .build(), new LambdaGauge(gcMetrics::getMaxDataSize));
```
Comment (geoand) [inline extensions/smallrye-metrics/runtime/src/main/java/io/quarkus/smallrye/metrics/runtime/SmallRyeMetricsRecorder.java]: While we are at it, can we remove the use of lambdas in runtime code please?

Comment (jmartisk) [reply extensions/smallrye-metrics/runtime/src/main/java/io/quarkus/smallrye/metrics/runtime/SmallRyeMetricsRecorder.java]: Do we really have to do that? 
see https://github.com/quarkusio/quarkus/pull/3732#issuecomment-526520841 
I'm not an expert on this topic but it seems the memory penalty is negligible

Comment (geoand) [reply extensions/smallrye-metrics/runtime/src/main/java/io/quarkus/smallrye/metrics/runtime/SmallRyeMetricsRecorder.java]: We don't **have** to do it, but it's been a general guideline for the runtime code of the project that we generally do try to follow.

Comment (jmartisk) [reply extensions/smallrye-metrics/runtime/src/main/java/io/quarkus/smallrye/metrics/runtime/SmallRyeMetricsRecorder.java]: Ok I'll do that then, but give me some time, hopefully will get round to it tomorrow :)

Comment (geoand) [reply extensions/smallrye-metrics/runtime/src/main/java/io/quarkus/smallrye/metrics/runtime/SmallRyeMetricsRecorder.java]: Sure thing, no rush :)

File: extensions/smallrye-metrics/runtime/src/main/java/io/quarkus/smallrye/metrics/runtime/MicrometerGCMetrics.java [hunk 2]
```diff
@@ -147,28 +148,34 @@ public void startWatchingNotifications() {
...
-                        MetricType.COUNTER,
-                        MetricUnits.NONE,
-                        "Time spent in GC pause",
-                        true,
-                        metricName.replace(".", "_") + "_seconds_count");
+                ExtendedMetadata countMetadata = new ExtendedMetadataBuilder()
+                        .withName(metricName + ".seconds.count")
+                        .withType(MetricType.COUNTER)
+                        .withUnit(MetricUnits.NONE)
+                        .withDescription("Time spent in GC pause")
```
Comment (gsmet) [inline extensions/smallrye-metrics/runtime/src/main/java/io/quarkus/smallrye/metrics/runtime/MicrometerGCMetrics.java]: Is it normal that both the descriptions of this one and the one above are identical?

File: extensions/smallrye-metrics/runtime/src/main/java/io/quarkus/smallrye/metrics/runtime/MicrometerGCMetrics.java [hunk 3]
```diff
@@ -147,28 +148,34 @@ public void startWatchingNotifications() {
...
-                        MetricType.COUNTER,
-                        MetricUnits.MILLISECONDS,
-                        "Time spent in GC pause",
-                        true,
-                        metricName.replace(".", "_") + "_seconds_sum"), tags).inc(duration);
+                registry.counter(new ExtendedMetadataBuilder()
+                        .withName(metricName + ".seconds.sum")
+                        .withType(MetricType.COUNTER)
+                        .withUnit(MetricUnits.MILLISECONDS)
+                        .withDescription("Time spent in GC pause")
```
Comment (gsmet) [inline extensions/smallrye-metrics/runtime/src/main/java/io/quarkus/smallrye/metrics/runtime/MicrometerGCMetrics.java]: And same here?

**Review by gsmet — DISMISSED** [review]
I added a small question. Feel free to dismiss if uncalled for as it's not really related to this change.

File: extensions/smallrye-metrics/runtime/src/main/java/io/quarkus/smallrye/metrics/runtime/MicrometerGCMetrics.java (hunk 2 above)
Comment (jmartisk) [reply extensions/smallrye-metrics/runtime/src/main/java/io/quarkus/smallrye/metrics/runtime/MicrometerGCMetrics.java]: I too find it a bit odd, but that's how Micrometer (in the latest version) does it, and we try to mimic it as closely as possible

**Review by gsmet — APPROVED** [review]

**Comment by gsmet** [comment]
Merged, thanks!

**Comment by jmartisk** [comment]
I was still planning to incorporate @geoand 's request but no problem, I'll open a new PR for it :)

**Comment by gsmet** [comment]
Yes, I decided to merge that one as the lambda thing is totally unrelated.

**Pull request merged**